austrakka/components/proforma/__init__.py

```diff
@@ -147,11 +147,22 @@ def proforma_attach(abbrev: str,
     required=False,
     help="Project to fill in on the Groups for Sharing tab",
 )
-def proforma_generate(abbrev: str, restrict, nndss, project):
+@click.option(
```

Comment:
I think we should always be defining options in `options.py` regardless if they're only used once.
Other than just having a single place for them so reuse actually occurs rather than adding duplicates down the line, the immediate reason is we actually use a custom class (`AusTrakkaCliOption`) for our options (I'm just remembering this now) which adds `[Accepts Multiple]` to the help output. If you check the help output for `--metadata-class` at the moment it does not state that it accepts multiple.

Reply:
I think there's two separate concerns here - one is reuse, and the other is our custom class behaviour. The custom class question is easy, it should always be used, wherever the option is defined, so one way or the other I should update this (although note we also have a MutuallyExclusiveOption class and these classes aren't related, that might need fixing).

At the moment the pattern across most of the CLI is that options only used locally are defined locally - e.g. see the rest of the proforma functions in this file. If we want to do this, we should do this everywhere.

I am a bit reluctant to put all single-use options in options.py however. One concern is that I think it actually decreases code clarity in that it's not obvious what an option's text and behaviour is - less of a concern for commonly used options which we can be expected to be familiar with. The other is that our global option namespace may simply not be big enough for all our options, unless we are not allowed short forms. For instance, here I am using `-c` for metadata class; I don't think this command is important enough to take -c out of the global namespace. At the same time, I don't want to make every parameter verbose. I'd say that single-use options ought not to be required to be different from other single-use options.

Reply:
Good point about not cluttering the global namespace. Let's make `austrakka.utils.options._create_option` public and use that for creating our options. I'm pretty certain this is a drop-in replacement for `click.option`